xbmc/GUIPassword.cpp

```diff
@@ -117,6 +117,79 @@ bool CGUIPassword::IsItemUnlocked(CFileItem* pItem, const std::string &strType)
   return true;
 }
 
+bool CGUIPassword::IsItemUnlocked(CMediaSource* pItem, const std::string &strType)
+{
+  const std::shared_ptr<CProfileManager> profileManager = CServiceBroker::GetSettingsComponent()->GetProfileManager();
+
+  // \brief Tests if the user is allowed to access the MediaSource
+  // \param pItem The share folder item to access
+  // \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
+  // \return If access is granted, returns \e true
+  if (profileManager->GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE)
+    return true;
+
+  while (pItem->m_iHasLock > 1)
+  {
+    std::string strLockCode = pItem->m_strLockCode;
+    std::string strLabel = pItem->strName;
```

Comment:
changed this from `std::string strLabel = pItem->GetLabel();`as GetLabel() is not available in CMediaSource